File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        q = [-x for x in stones]
        heapq.heapify(q)
        while len(q) > 1:
            f = heapq.heappop(q)
            s = heapq.heappop(q)
            if(f != s):
                heapq.heappush(q, f-s)
        return 0 if len(q) == 0 else -q[0]
            
        
            
        

File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> q = new PriorityQueue<>((a, b) -> b-a);
        for(int i=0;i<stones.length;i++){
            q.add(stones[i]);
        }
        while(q.size() > 1){
            int y = q.poll();
            int x = q.poll();
            if(x == y) continue;
            q.add(y-x);
        }
        return q.isEmpty() ? 0 : q.poll();
    }
}